### ISSUE TITLE
Examples: amp-sidebar trigger cannot be used with a keyboard

### ISSUE THREAD

**Issue by TetraLogicalHelpdesk** [ampproject/amphtml#31162]
## WCAG Level
Level AA

## Priority
Critical

## Type
Interaction

## Pages/screens/components affected
* [amp-sidebar in example Product Browser Page](https://amp.dev/documentation/examples/e-commerce/product_browse_page/)

## Description
Good design means supporting different modes of interaction, because there are many reasons why someone might use a keyboard instead of a mouse, and if keyboard interaction is not fully supported the website is effectively unusable.

In the [header with search](https://amp.dev/documentation/examples/e-commerce/product_browse_page/?format=websites#header-with-search) example of the [Product Browser Page](https://amp.dev/documentation/examples/e-commerce/product_browse_page/) demo page, the trigger to open the sidebar is currently not focusable/operable with the keyboard.

## User impact
Someone with mobility problems will often use a keyboard instead of a mouse to navigate a web page. A screen reader user will also tend to use a keyboard if they cannot see the screen to use the mouse. Therefore, if the website does not properly support keyboard interaction a large number of people are effectively prevented from using it.

In this instance, keyboard users will not be able to trigger the sidebar.

## Required solution
Make sure that all interactive controls can be used with a keyboard.

This solution must be applied to all instances of the issue identified within the test sample, then applied to all other instances of the same issue identified throughout the rest of the website.

## Implementation guidance

Currently, the trigger element for the sidebar uses an `<a>` element without an `href` attribute. These anchor elements don't receive keyboard focus by default

![The 'header with search' example, with the anchor element highlighted](https://user-images.githubusercontent.com/54933393/99181419-55ffcc80-2726-11eb-9213-46d0a89b374d.png)

```
<a id="sample-menu" on="tap:drawermenu.toggle">
  <amp-img ... alt="navigation" ...> ... </amp-img>
</a>
```

We recommend changing this control to a `<button>` element, which is structurally more appropriate (as it triggers in-page behaviour), and receives keyboard focus by default.

```
<button id="sample-menu" on="tap:drawermenu.toggle">
  <amp-img ... alt="navigation" ...> ... </amp-img>
</button>
```

## Test procedure(s)
Use these steps to confirm that the solution has been correctly applied to issues identified within the test sample, and to test the rest of the website for instances of the same issue:
1. Check that all interactive elements can be navigated to using the `Tab` key.
2. Check that all interactive elements can be activated using the `Enter` or `Space` keys.
3. Check that interactive elements can be interacted with using the arrow keys.

### Definition of done
Complete all of these tasks before closing this issue or indicating it is ready for retest:
* All issues identified within the test sample have been resolved.
* The rest of the website  has been tested for the same issue.
* All issues identified throughout the rest of the website  have been resolved or filed as new issues.

## Related standards
* [WCAG 2.1 Success Criterion 2.1.1 Keyboard (Level A)](https://www.w3.org/TR/WCAG21/#keyboard)

## More information
* [WebAIM: Keyboard](https://webaim.org/techniques/keyboard/)

## Test data
Test date: November 2020
